stubs/jsonschema/jsonschema/validators.pyi

```diff
@@ -94,15 +106,16 @@ class RefResolver:
     @property
     def resolution_scope(self): ...
     @property
-    def base_uri(self): ...
+    def base_uri(self) -> str: ...
     @contextmanager
+    @deprecated("jsonschema.RefResolver.in_scope is deprecated and will be removed in a future release.")
     def in_scope(self, scope) -> Generator[None]: ...
     @contextmanager
-    def resolving(self, ref) -> Generator[Incomplete, None, None]: ...
-    def resolve(self, ref): ...
-    def resolve_from_url(self, url): ...
-    def resolve_fragment(self, document, fragment): ...
-    def resolve_remote(self, uri): ...
+    def resolving(self, ref: str) -> Generator[Incomplete]: ...
+    def resolve(self, ref: str) -> tuple[str, Incomplete]: ...
+    def resolve_from_url(self, url: str): ...
+    def resolve_fragment(self, document, fragment: str): ...
+    def resolve_remote(self, uri: str): ...
 
 def validate(instance: object, schema: Schema, cls: type[_Validator] | None = None, *args: Any, **kwargs: Any) -> None: ...
-def validator_for(schema: Schema | bool, default=...): ...
+def validator_for(schema: Schema | bool, default: type[_Validator] | Unset = ...) -> type[_Validator]: ...
```

Comment:
Looking at the primer output, we probably need to to `._protocols.Validator` here, like the implementation does.